### ISSUE TITLE
Hot reloading functional components doesn't re-render them in place

### ISSUE THREAD

**Issue by jangarcia** [vue-styleguidist/vue-styleguidist#354]
**Current behavior**

Hot reloading works for functional components as well, but unlike non-functional components, template changes in single file functional components don't trigger a re-rendering of the preview until you change the page/route and then return. On the other hand, changes in the style or docs  block immediately update the preview.

**Expected behavior**

Functional components should be re-rendered in place after hot module replacement.

**Comment by elevatebart** [comment]
thank you @jangarcia, this was a hard one to catch.

**Comment by elevatebart** [comment]
The [Preview](https://github.com/vue-styleguidist/vue-styleguidist/blob/dev/packages/vue-styleguidist/src/rsg-components/Preview/Preview.js) component is responsible for rendering examples.

If you want to try your hand at it. If not I'll look into it ASAP.

**Comment by jangarcia** [comment]
Hi @elevatebart, I tried to get to the bottom of the problem, but unfortunately I didn't get very far, so I would leave it up to you. Thanks!

**Comment by Stegosource** [comment]
I don't think this is an issue with Vue Styleguidist, as much as it is with functional components. I notice a lot of the time with just regular Vue development that functional components do no trigger a hot reload when the content has changed.

**Comment by elevatebart** [comment]
It would probably make sense since they do not have an actual existence in the virtual dom.
Maybe I can wrap them in a real component that would force re-loading ;)
